src/containers/pages/IRS/JurisdictionsReport/helpers.ts

```diff
@@ -0,0 +1,308 @@
+    columns: [
+      {
+        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
+        Header: '',
+        accessor: 'roomcov',
+        className: 'indicator centered',
+      },
+    ],
+  },
+];
+
+/** columns for Namibia IRS focus (spray) areas */
+export const ZambiaFocusAreasColumns = [
+  {
+    Header: 'Name',
+    columns: [
+      {
+        Header: '',
+        accessor: 'jurisdiction_name',
+        minWidth: 180,
+      },
+    ],
+  },
+  {
+    Header: 'Structures on the ground',
+    columns: [
+      {
+        Header: '',
+        accessor: 'totstruct',
+      },
+    ],
+  },
+  {
+    Header: 'Found',
+    columns: [
+      {
+        Header: '',
+        accessor: 'foundstruct',
+      },
+    ],
+  },
+  {
+    Header: 'Sprayed',
+    columns: [
+      {
+        Header: '',
+        accessor: 'sprayedstruct',
+      },
+    ],
+  },
+  {
+    Header: 'Spray Coverage (Effectiveness)',
+    columns: [
+      {
+        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
+        Header: '',
+        accessor: 'spraycov',
+        className: 'indicator centered',
+      },
+    ],
+  },
+  {
+    Header: 'Found Coverage',
+    columns: [
+      {
+        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
+        Header: '',
+        accessor: 'spraytarg',
+        className: 'indicator centered',
+      },
+    ],
+  },
+  {
+    Header: 'Spray Success Rate (PMI SC)',
+    columns: [
+      {
+        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
+        Header: '',
+        accessor: 'spraysuccess',
+        className: 'indicator centered',
+      },
+    ],
+  },
+  {
+    Header: 'Room coverage of structures sprayed',
+    columns: [
+      {
+        Cell: (cell: CellInfo) => getIRSThresholdAdherenceIndicator(cell),
+        Header: '',
+        accessor: 'roomcov',
+        className: 'indicator centered',
+      },
+    ],
+  },
+];
+
+export const IRSTableColumns: { [key: string]: Column[] } = {
+  namibia2019: NamibiaColumns,
+  zambiaFocusArea2019: ZambiaFocusAreasColumns,
+  zambiaJurisdictions2019: ZambiaJurisdictionsColumns,
+};
+
+/** extract hierarchical tree of jurisdictions as a list, with the highest
+ * level item in the hierarchy as the 0th element
+ * @param {FlexObject[]} tree - the data set
+ * @param {FlexObject} item - the item in question
+ * @returns {FlexObject[]} - the hierarchy for the item in question
+ */
+export function getTree(
+  tree: FlexObject[],
+  item: FlexObject,
+  result: FlexObject[] = []
+): FlexObject[] {
+  if (item.jurisdiction_parent_id === null && !result.includes(item)) {
+    result.push(item);
+  } else {
+    const parentItems = tree.filter(el => el.jurisdiction_id === item.jurisdiction_parent_id);
+    if (parentItems.length > 0) {
+      if (!result.includes(parentItems[0])) {
+        result.push(parentItems[0]);
+      }
+      return getTree(tree, parentItems[0], result);
```

Comment:
this is good for now, and this function looks tail recursive which is good, but if the javascript interpreter isn't supporting tail call optimization, for large hierarchies this could cause memory issues by filling the stack as it's doing the recursion so we may want to rewrite as explicitly iterative in the future to avoid this